### ISSUE TITLE
Invoking parinfer in normal mode

### ISSUE THREAD

**Issue by spinningarrow** [bhurlow/vim-parinfer#24]
I was wondering if there's any support for invoking parinfer in normal mode; for instance, if I put the cursor over a `(` and press `x`, only that paren gets deleted. If I then go into Insert mode and immediately leave, the matching `)` gets deleted automatically by parinfer. Is there going to be support for that happening without jumping in and out of Insert mode?

**Comment by tomaskulich** [comment]
+1

**Comment by bhurlow** [comment]
@spinningarrow I think the simplest thing here might just be to remap `x` to call a function that deletes a character and then call parinfer to rebalance. Took a pass at this, looks promising

**Comment by tomaskulich** [comment]
The behavior of the plugin is too buggy for me. I created an infinitely simplified version of it in case anyone is interested: https://github.com/tomaskulich/vim-parinfer-bare

**Comment by spinningarrow** [comment]
@tomaskulich Could you describe some of the buggy behaviour? Not disagreeing with you per se; I'm currently using LightTable for clojure as I found parinfer in vim not as pleasing but I think it'll help improve this plugin if there are specific things to address.

**Comment by bhurlow** [comment]
@spinningarrow the buggy behavior is primary related to: https://github.com/bhurlow/vim-parinfer/issues/32, and it's actually pretty debilitating. Are you using a parinfer implementation in LightTable? Can you elaborate on how they're wired differently?

**Comment by spinningarrow** [comment]
> Are you using a parinfer implementation in LightTable?

I'm using this one: https://github.com/mauricioszabo/lt_parinfer. I think it's the only one available (or at least was when I installed it).

I haven't looked that deeply into how they're wired differently, but I don't have any complaints with it so far. The problem for me in vim was that I make a lot of changes (especially deletions) in normal mode and vim-parinfer only works in insert mode. I'm still only tinkering with clojure though, which is probably why LightTable hasn't gotten to me yet (I prefer vim for everything else).

**Comment by tomaskulich** [comment]
For example take this wana-be-form:

```
(fn [] a
  (fn [] b
    (fn [] c
```

Now when parinfer is invoked, only first line (!) is sent to the viml-parinfer for analysis. Therefore, the result is:

```
(fn [] a)
  (fn [] b
    (fn [] c
```

The funny (but understandable) thing is that when I now invoke the parinfer (by switching to insert mode and back), what I got is:
```
(fn [] a
  (fn [] b))
    (fn [] c
```

So the parinfer indent mode is not even behaving idempotently!

**Comment by bhurlow** [comment]
it wants to be a form so bad! :)

**Comment by bhurlow** [comment]
@spinningarrow these changes: https://github.com/bhurlow/vim-parinfer/blob/master/plugin/parinfer.vim#L123-L124 override `dd` and `p` respectively so you can delete and paste in normal mode and have parfiner rebalance the form. Are there other ops you'd like to see happen in normal mode?

**Comment by spinningarrow** [comment]
Ideally I'd like any text-modifying operation to invoke parinfer seamlessly. I've been using parinfer in LightTable with vim mode enabled, and it's quite a breeze. (Relatedly, what would it take to get parinfer to work in insert mode?)

**Comment by spinningarrow** [comment]
Closed by #45 and #46.
